### DISCUSSION TITLE
Opening the budibase app redirected from another app/qr code. Access url data to populate app fields

### DISCUSSION THREAD

**Discussion by programmeddeath1** [Budibase/budibase#5157] [Help]
I wanted to know if this was possible through budibase just exploring the feasibility for a business app.
Is it possible to redirect from an existing application to a budibase app with parameters in the url being read before the app screen loads and populating that information on the screen in the app? sort of like getting the url in javascript and parsing and populating that info to the app fields.

I read another discussion that gives a workaround for accessing url variables, but will I be able to run this before page load on the app?

Thank you for your help in advance.

**Comment by mjashanks** [comment]
Budibase directly supports URL **path** parameters.

When you create a screen, you can declare its URL as `/customer/:id`, so that `/customer/1` will load the customer record. There's no reason why you couldn't do `/customer/:firstname/:lastname/:email` and use these parameters in the same way. 

However, a few things on this
1. You can see that this might get messy with lots of parameters
2. When browsing to the URL, you need to submit values for **all** of the parameters. 

So, if you didn't want to supply the :email parameter (as above), you would need to do `/customer/michael/shanks/default`, then handle `default` appropriately

**Comment by melohagan** [comment]
This may also help:

![Screenshot 2022-03-30 at 12 25 52](https://user-images.githubusercontent.com/101575380/160824016-297a482a-d63d-430c-be33-b400106fb675.png)

![Screenshot 2022-03-30 at 09 28 57](https://user-images.githubusercontent.com/101575380/160787221-14ccc589-2f8a-428a-974b-830150807346.png)

Under the default value of each of your form fields you can specify the following JavaScript:
```
function getQueryVar(variable) {
	let query = $("URL.query");
	if (!query) return "";
	let vars = query.split('&');
    for (let i = 0; i < vars.length; i++) {
			let pair = vars[i].split('=');
      if (decodeURIComponent(pair[0]) == variable) {
          return decodeURIComponent(pair[1]);
      }
    }
}
return getQueryVar('Name');
```
